common/static/common/js/discussion/discussion_router.js

```diff
@@ -102,11 +108,43 @@
             };
 
             DiscussionRouter.prototype.showThread = function(forum_name, thread_id) {
+                var self, callback;
+                self = this;
                 this.thread = this.discussion.get(thread_id);
+
+                if (!this.thread){
+                    callback = function(thread) {
+                        self.thread = thread;
+                        self.renderThreadView();
+                    };
+                    this.retrieveSingleThread(forum_name, thread_id, callback);
+                }
+                else {
+                    this.renderThreadView()
+                }
+            };
+
+            DiscussionRouter.prototype.retrieveSingleThread = function(forum_name, thread_id, callback) {
+                DiscussionUtil.safeAjax({
+                    url: DiscussionUtil.urlFor('retrieve_single_thread', forum_name, thread_id),
+                    success: function(data) { callback(new Thread(data['content'])); },
+                    error: function(xhr) {
+                        var errorMessage = (xhr.status === 404) ?
+                            gettext("The thread you selected has been deleted. Please select another thread.") :
```

Comment:
@bradenmacdonald [yes](https://github.com/edx/edx-platform/blob/4fd7f2936aefe9421c704c266d3f35bb3ca7f1a5/lms/djangoapps/django_comment_client/forum/views.py#L323).

Well, in theory, it will also happen if discussions are disabled, but we won't get to this code than in the first place. Another option is changed URL, but that would make discussions totally broken. So, yes, there's only one legitimate condition for 404 - deleted thread